aggregation/src/lib.rs

```diff
@@ -0,0 +1,495 @@
+    /// verify the root by making an assumption (i.e. by calling `env::verify`), and verifies in an
+    /// EVM context may reference previously proven claims via a verification cache in storage.
+    pub root: Option<Receipt>,
+
+    /// Merkle proof for inclusion in the set of claims attested to by the root receipt.
+    pub merkle_path: Vec<Digest>,
+
+    /// A digest of the verifier parameters that can be used to verify this receipt.
+    ///
+    /// Acts as a fingerprint to identity differing proof system or circuit versions between a
+    /// prover and a verifier. Is not intended to contain the full verifier parameters, which must
+    /// be provided by a trusted source (e.g. packaged with the verifier code).
+    pub verifier_parameters: Digest,
+}
+
+impl<Claim> SetInclusionReceipt<Claim>
+where
+    Claim: Digestible + Debug + Clone + Serialize,
+{
+    /// Construct a [SetInclusionReceipt] with the given Merkle inclusion path and claim.
+    ///
+    /// Path should contain all sibling nodes in the tree from the leaf to the root. Note that the
+    /// path does not include the leaf or the root itself. Resulting receipt will have default
+    /// verifier paramaters and no root receipt.
+    #[cfg(feature = "verify")]
+    pub fn from_path(claim: impl Into<MaybePruned<Claim>>, merkle_path: Vec<Digest>) -> Self {
+        Self {
+            claim: claim.into(),
+            root: None,
+            merkle_path,
+            verifier_parameters: SetInclusionReceiptVerifierParameters::default()
+                .digest::<sha::Impl>(),
+        }
+    }
+
+    /// Add the given root receipt to this set inclusion receipt.
+    ///
+    /// See [SetInclusionReceipt::root] for more information about the root receipt.
+    pub fn with_root(self, root_receipt: Receipt) -> Self {
+        Self {
+            root: Some(root_receipt),
+            ..self
+        }
+    }
+
+    /// Drops the root receipt from this [SetInclusionReceipt].
+    ///
+    /// This is useful when the verifier has a cache of verified roots, as is the case for smart
+    /// contract verifiers. Use this method when submitting this receipt as part of a batch of
+    /// receipts to be verified, to reduce the encoded size of this receipt.
+    pub fn without_root(self) -> Self {
+        Self { root: None, ..self }
+    }
+
+    /// Verify the integrity of this receipt, ensuring the claim is attested to by the seal.
+    #[cfg(feature = "verify")]
+    pub fn verify_integrity(&self) -> Result<(), anyhow::Error> {
+        self.verify_integrity_with_context(
+            &VerifierContext::default(),
+            SetInclusionReceiptVerifierParameters::default(),
+            Some(RecursionVerifierParamters::default()),
+        )
+    }
+
+    /// Verify the integrity of this receipt, ensuring the claim is attested to by the seal.
+    // TODO: Use a different error type (e.g. the one from risc0-zkvm).
+    pub fn verify_integrity_with_context(
```

Comment:
It's not really required, so I opted not to. That feature mostly controls whether or not to include the guest build info for e.g. image IDs. It exists to avoid a circular dependency, since the guest depends on this crate. I named it `verify` because the main reason you'd turn the feature on is to have access to the `.verify()` function. I failed to think of a name that was more accurate without being confusing.

Reply:
Ended up moving all of this under the `verify` feature 👍 
3ae5e230be1a38d4ad1583ff707924ed5764086a